productivity-suite/app/legacy-app/src/auth.tsx

```diff
@@ -3,44 +3,52 @@ import { createContext, useContext, useEffect, useState } from "react";
 import { Navigate, useLocation, useNavigate } from "react-router-dom";
 import {
   deleteCurrentUser,
+  getCurrentUser,
   getUserData,
   saveCurrentUser,
   setUserData,
 } from "shared";
 import { initialPlaceholderTodoList } from "./initialPlaceholderTodoList";
 
-interface AuthContextType {
-  user: string | null;
-  login: (userName: string) => void;
-  logout: () => void;
-}
+export const AUTH_LOADING = Symbol("Auth Loading");
+export type AuthContextType = { user: string | null } | undefined;
 
 let AuthContext = createContext<AuthContextType>(null!);
```

Comment:
with the current changes is `null!` still needed?

Reply:
ah, I think it is, but it should be:
```suggestion
let AuthContext = createContext<AuthContextType>(undefined);
```
I think?

Reply:
I think so. Thanks